perses/rjmc/geometry.py

```diff
@@ -96,46 +97,49 @@ def __init__(self, metadata=None, use_sterics=False, verbose=True):
         self.verbose = verbose
         self.use_sterics = use_sterics
 
-    def propose(self, top_proposal, current_positions, beta):
+    def propose(self, top_proposal, current_sampler_state, beta):
         """
         Make a geometry proposal for the appropriate atoms.
 
         Arguments
         ----------
         top_proposal : TopologyProposal object
             Object containing the relevant results of a topology proposal
+        current_sampler_state : openmmtools.states.SamplerState
+            The current positions and box vectors of the proposal
         beta : float
             The inverse temperature
 
         Returns
         -------
-        new_positions : [n, 3] ndarray
+        new_sampler_state : openmmtools.states.SamplerState
             The new positions of the system
         logp_proposal : float
             The log probability of the forward-only proposal
         """
+        current_positions = current_sampler_state.positions
         current_positions = current_positions.in_units_of(units.nanometers)
         if not top_proposal.unique_new_atoms:
             structure = parmed.openmm.load_topology(top_proposal.old_topology, top_proposal.old_system)
             atoms_with_positions = [structure.atoms[atom_idx] for atom_idx in top_proposal.new_to_old_atom_map.keys()]
             new_positions = self._copy_positions(atoms_with_positions, top_proposal, current_positions)
             return new_positions, 0.0
-        logp_proposal, new_positions = self._logp_propose(top_proposal, current_positions, beta, direction='forward')
+        logp_proposal, new_positions = self._logp_propose(top_proposal, current_sampler_state, beta, direction='forward')
         self.nproposed += 1
-        return new_positions, logp_proposal
+        return states.SamplerState(new_positions, box_vectors=current_sampler_state.box_vectors), logp_proposal
 
 
-    def logp_reverse(self, top_proposal, new_coordinates, old_coordinates, beta):
+    def logp_reverse(self, top_proposal, new_sampler_state, old_sampler_state, beta):
         """
         Calculate the logp for the given geometry proposal
 
         Arguments
         ----------
         top_proposal : TopologyProposal object
             Object containing the relevant results of a topology proposal
-        new_coordinates : [n, 3] np.ndarray
+        new_sampler_state : openmmtools.states.SamplerState
             The coordinates of the system after the proposal
-        old_coordiantes : [n, 3] np.ndarray
+        old_sampler_state : openmmtools.states.SamplerState
```

Comment:
Again, `beta` below should be a unit-bearing quantity.

Reply:
It is, I think I just need to fix the docstring.